src/negativity.jl

```diff
@@ -0,0 +1,136 @@
+@doc raw"""
+    negativity(ρ::QuantumObject, subsys::Int; logarithmic::Bool=false)
+
+Compute the [negativity](https://en.wikipedia.org/wiki/Negativity_(quantum_mechanics)) ``N(\rho) = \frac{\Vert \rho^{\Gamma}\Vert_1 - 1}{2}``  
+where ``\rho^{\Gamma}`` is the partial transpose of ``\rho`` with respect to the subsystem,  
+and ``\Vert X \Vert_1=\Tr\sqrt{X^\dagger X}`` is the trace norm.
+
+# Arguments
+- `ρ::QuantumObject`: The density matrix (`ρ.type` must be `OperatorQuantumObject`).
+- `subsys::Int`: an index that indicates which subsystem to compute the negativity for.
+- `logarithmic::Bool`: choose whether to calculate logarithmic negativity or not. Default as `false`
+
+# Returns
+- `N::Real`: The value of negativity.
+
+# Examples
+
+```
+julia> Ψ = 1 / √2 * ( basis(2,0) ⊗ basis(2,0) + basis(2,1) ⊗ basis(2,1) )
+Quantum Object:   type=Ket   dims=[2, 2]   size=(4,)
+4-element Vector{ComplexF64}:
+ 0.7071067811865475 + 0.0im
+                0.0 + 0.0im
+                0.0 + 0.0im
+ 0.7071067811865475 + 0.0im
+
+julia> ρ = ket2dm(Ψ)
+Quantum Object:   type=Operator   dims=[2, 2]   size=(4, 4)   ishermitian=true
+4×4 Matrix{ComplexF64}:
+ 0.5+0.0im  0.0+0.0im  0.0+0.0im  0.5+0.0im
+ 0.0+0.0im  0.0+0.0im  0.0+0.0im  0.0+0.0im
+ 0.0+0.0im  0.0+0.0im  0.0+0.0im  0.0+0.0im
+ 0.5+0.0im  0.0+0.0im  0.0+0.0im  0.5+0.0im
+
+julia> negativity(ρ, 2)
+0.4999999999999998
+```
+"""
+function negativity(ρ::QuantumObject, subsys::Int; logarithmic::Bool=false)
+    mask = fill(false, length(ρ.dims))
+    try
+        mask[subsys] = true
+    catch
+        error("Invalid index of subsys: $subsys")
+    end
+
+    ρ_pt = partial_transpose(ρ, mask)
+    tr_norm = sum(sqrt.(eigvals(ρ_pt' * ρ_pt)))
```

Comment:
Here maybe `sum(sqrt, eigvals(ρ_pt' * ρ_pt))` might be better, avoiding extra allocations.

Reply:
Thank you for the suggestion
I have made an update in commit cd7d7a79d0fca3cbf6ef9c85877066be09f960d4